### PR TITLE
More BehaviorService related code cleanup

### PR DESCRIPTION
**Pull request by JeremyKuhne**
- Use null coalescing operators
- Use conditional operators
- Few more if block simplifications
- Minor comment and spacing cleanup
- Simplify usings

Not exhaustive, just grabbing low-hanging fruit

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/3911)

### CODE REVIEW

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/BehaviorService.AdornerWindow.MouseHook.cs [hunk 1]
```diff
@@ -78,45 +78,47 @@ private void HookMouse()
...
+                                throw;
                             }
                         }
+                        finally
+                        {
+                            _currentAdornerWindow = null;
+                        }
                     }
 
                     Debug.Assert(_isHooked, "How did we get here when we are diposed?");
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/BehaviorService.AdornerWindow.MouseHook.cs:124]: typo - disposed

Comment (JeremyKuhne) [reply src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/BehaviorService.AdornerWindow.MouseHook.cs:124]: I only changed what caught my eye as I went through. :) I'll fix this on the next PR.

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ControlDesigner.cs [hunk 2]
```diff
@@ -295,9 +271,10 @@ private bool IsResizableConsiderAutoSize(PropertyDescriptor autoSizeProp, Proper
             bool resizable = true;
             bool autoSize = false;
             bool growOnly = false;
-            if (autoSizeProp != null &&
-                !(autoSizeProp.Attributes.Contains(DesignerSerializationVisibilityAttribute.Hidden) ||
-                  autoSizeProp.Attributes.Contains(BrowsableAttribute.No)))
+
+            if (autoSizeProp != null
+                && !(autoSizeProp.Attributes.Contains(DesignerSerializationVisibilityAttribute.Hidden)
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ControlDesigner.cs:276]: Consider factoring `autoSizeProp.Attributes` into a local variable.

**Review by Tanya-Solyanik — DISMISSED** [review]

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/BehaviorService.AdornerWindow.MouseHook.cs (hunk 1 above)
Comment (JeremyKuhne) [reply src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/BehaviorService.AdornerWindow.MouseHook.cs:124]: I'll get this one in this PR- I accidentally used `is not null` and need to undo that :)

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ControlDesigner.cs (hunk 2 above)
Comment (JeremyKuhne) [reply src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ControlDesigner.cs:276]: It is sort of better with it? Still squinting at it, but I'll put this in.

**Review by Tanya-Solyanik — APPROVED** [review]

**Pull request merged**